rootston/seat.c

```diff
@@ -306,6 +301,28 @@ struct roots_seat *roots_seat_create(struct roots_input *input, char *name) {
 	return seat;
 }
 
+static void seat_update_capabilities(struct roots_seat *seat) {
+	uint32_t caps = 0;
+	if (!wl_list_empty(&seat->keyboards)) {
+		caps |= WL_SEAT_CAPABILITY_KEYBOARD;
+	}
+	if (!wl_list_empty(&seat->pointers) || !wl_list_empty(&seat->tablet_tools)) {
+		caps |= WL_SEAT_CAPABILITY_POINTER;
+	}
+	if (!wl_list_empty(&seat->touch)) {
+		caps |= WL_SEAT_CAPABILITY_TOUCH;
+	}
+	wlr_seat_set_capabilities(seat->seat, caps);
+
+	// Hide cursor if seat doesn't have pointer or touch capability
+	if ((caps & (WL_SEAT_CAPABILITY_POINTER | WL_SEAT_CAPABILITY_TOUCH)) == 0) {
```

Comment:
I think you can hide the cursor if the seat has the touch capability but no pointer right? I think touch points in the current implementation never move the cursor, but I'd have to recheck.

Reply:
Oh, you're right. I checked that but incorrectly read `roots_cursor_press_button`.